### PR TITLE
Query movie with AppAccess

### LINKED ISSUE #2657: Queries with AppAccess of movie
Update the query so the films are loaded correctly thanks to the appAcces of the movie

### PR DESCRIPTION
**Pull request by Mussuranae**
Call movies in dashboard and marketplace with the appAccess status, to filter the app on which movie can be and not
Fix #2657

### CODE REVIEW

File: apps/backend-functions/src/internals/algolia.ts [hunk 1]
```diff
@@ -95,6 +96,9 @@ export function storeSearchableMovie(
       budget: movie.budget.totalBudget?.amount || movie.budget.estimatedBudget?.from || 0,
       orgName,
       storeType: movie.main.storeConfig?.storeType || '',
+      appAccess: !! movie.main.storeConfig!.appAccess ?
```
Comment (Hugo-Hontans) [inline apps/backend-functions/src/internals/algolia.ts:99]: Don't put a space between !! and movie :)

File: apps/catalog/catalog/src/app/guards/catalog-app.guard.ts [hunk 2]
```diff
@@ -15,9 +15,9 @@ export class CatalogAppGuard implements CanActivate {
     const isMarketplace = state.url.split('/').includes('marketplace');
     const org = this.query.getActive();
     if (isMarketplace) {
-      return org.appAccess.catalog.marketplace ? true : this.router.parseUrl('c/o/dashboard');
+      return org.appAccess.catalog.marketplace ? true : this.router.parseUrl('c/o/marketplace');
```
Comment (Hugo-Hontans) [inline apps/catalog/catalog/src/app/guards/catalog-app.guard.ts:18]: If it's marketplace, the guard return true, if not it should return on dashboard no ? I don't understand your change

File: apps/catalog/catalog/src/app/guards/catalog-app.guard.ts [hunk 3]
```diff
@@ -15,9 +15,9 @@ export class CatalogAppGuard implements CanActivate {
...
     } else {
-      return org.appAccess.catalog.dashboard ? true : this.router.parseUrl('c/o/marketplace');
+      return org.appAccess.catalog.dashboard ? true : this.router.parseUrl('c/o/dashboard');
```
Comment (Hugo-Hontans) [inline apps/catalog/catalog/src/app/guards/catalog-app.guard.ts:20]: same

File: libs/movie/src/lib/movie/guards/movie-active.guard.ts [hunk 4]
```diff
@@ -4,19 +4,33 @@ import { MovieState } from "../+state/movie.store";
...
   // Sync and set active
   sync(next: ActivatedRouteSnapshot) {
     return this.service.syncActive({ id: next.params.movieId }).pipe(
-      map(movie =>  movie ? true : (this.redirect || next.data.redirect))
+      map(movie => {
+        if (!!movie) {
+          const appName = this.routerQuery.getValue().state.root.data.app;
+          return haveAppAccess(appName, movie.main.storeConfig.appAccess);
+        } else {
+          return this.redirect || next.data.redirect;
```
Comment (Hugo-Hontans) [inline libs/movie/src/lib/movie/guards/movie-active.guard.ts:30]: Can you check the issue about @CollectionGuardConfig, we miss parameters on it for the redirect here. Ask to François I think it's quick

File: apps/catalog/catalog/src/app/guards/catalog-app.guard.ts (hunk 2 above)
Comment (Mussuranae) [reply apps/catalog/catalog/src/app/guards/catalog-app.guard.ts:18]: Yeah sure, it looks like I wasn't really awake this morning

File: apps/backend-functions/src/internals/algolia.ts [hunk 5]
```diff
@@ -95,6 +96,9 @@ export function storeSearchableMovie(
       budget: movie.budget.totalBudget?.amount || movie.budget.estimatedBudget?.from || 0,
       orgName,
       storeType: movie.main.storeConfig?.storeType || '',
+      appAccess: !!movie.main.storeConfig!.appAccess ?
```
Comment (GrandSchtroumpf) [inline apps/backend-functions/src/internals/algolia.ts:99]: It's too difficult to read. Create a variable for that above.

File: apps/catalog/catalog/src/app/dashboard/pages/home/home.component.ts [hunk 6]
```diff
@@ -13,7 +13,7 @@ import { MovieService } from '@blockframes/movie/+state/movie.service';
 })
 export class HomeComponent implements OnInit, OnDestroy {
   public movieAnalytics$: Observable<MovieAnalytics[]>;
-  public movies$ = this.movieQuery.selectAll({ filterBy: movie => movie.main.storeConfig.status === 'accepted' });
+  public movies$ = this.movieQuery.selectAll({ filterBy: movie => movie.main.storeConfig.status === 'accepted' && movie.main.storeConfig.appAccess.catalog === true });
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/pages/home/home.component.ts:16]: Put it on several lines : 
```typescript
const isAcceptedInApp = movie => movie.main.storeConfig.status === 'accepted' & movie.main.storeConfig.appAccess.catalog
this.movieQuery.selectAll({ filterBy: isAcceptedInApp })
```

File: apps/catalog/catalog/src/app/dashboard/title/list/list.component.ts [hunk 7]
```diff
@@ -80,7 +80,7 @@ export class TitleListComponent implements OnInit, OnDestroy {
     const movies$ = this.filter$.pipe(
       switchMap(filter =>
         this.query.selectAll({
-          filterBy: movie => (filter ? movie.main.storeConfig.storeType === filter : true)
+          filterBy: movie => (filter ? movie.main.storeConfig.storeType === filter : true && movie.main.storeConfig.appAccess.catalog === true)
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/title/list/list.component.ts:83]: Same here

File: libs/movie/src/lib/movie/guards/movie-active.guard.ts (hunk 4 above)
Comment (Mussuranae) [reply libs/movie/src/lib/movie/guards/movie-active.guard.ts:35]: It's ok, I understood how it works. The redirection is working now.

File: libs/movie/src/lib/movie/guards/movie-active.guard.ts [hunk 8]
```diff
@@ -4,19 +4,38 @@ import { MovieState } from "../+state/movie.store";
...
 
   // Sync and set active
   sync(next: ActivatedRouteSnapshot) {
     return this.service.syncActive({ id: next.params.movieId }).pipe(
-      map(movie =>  movie ? true : (this.redirect || next.data.redirect))
+      map(movie => {
+        if (!!movie) {
+          const appName = this.routerQuery.getValue().state.root.data.app;
+          const hasAccess = hasAppAccess(appName, movie.main.storeConfig.appAccess);
+          if (hasAccess === true) {
```
Comment (Hugo-Hontans) [inline libs/movie/src/lib/movie/guards/movie-active.guard.ts:29]: hasAccess is all time a boolean so you can just use `if (hasAccess)`

File: apps/festival/festival/src/app/dashboard/title/list/list.component.ts [hunk 9]
```diff
@@ -74,6 +75,7 @@ export class ListComponent implements OnInit, OnDestroy {
 
     const titles$ = this.orgQuery.selectActive().pipe(
       switchMap(org => this.service.valueChanges(org.movieIds)),
+      map(movies => movies.filter(movie => movie.main.storeConfig.appAccess.festival === true))
```
Comment (Hugo-Hontans) [inline apps/festival/festival/src/app/dashboard/title/list/list.component.ts:78]: just return `movie.main.storeConfig.appAccess.festival`

**Pull request merged**